cmd/maya-exporter/app/collector/jivacollector.go

```diff
@@ -183,11 +187,26 @@ func (e *VolumeExporter) Describe(ch chan<- *prometheus.Desc) {
 // concurrent readers.
 
 // Collect collects all the registered stats metrics from the OpenEBS volumes.
+// It tries to reconnect with the volume if there is any error via a goroutine.
 func (e *VolumeExporter) Collect(ch chan<- prometheus.Metric) {
-	if err := e.collect(); err != nil {
-		return
+	// verify if controller url is not empty
+	if len(e.VolumeControllerURL) != 0 {
+		// collect the metrics from jiva controller and send it via channels
+		if err := e.collect(); err != nil {
```

Comment:
The method name is collect. Should this be collectAndSend?
I am finding it difficult to relate the comment & the code.

Reply:
if you look at the code given below to this, we are sending the metrics via channels (i.e, collect -> collect the metrics and set the values to the respected fields and then it sends the value to the prometheus via channels.)